### PR TITLE
add access authorities check for assignedDataSet api.

### PR DESCRIPTION
**Pull request by vietnguyen**

### CODE REVIEW

File: dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FormUtils.java [hunk 1]
```diff
@@ -154,6 +154,11 @@ private static CategoryCombo getCategoryCombo( DataSet dataset, Set<Organisation
             {
                 for ( DataElementCategory cat : cats )
                 {
+                    if ( cat.getAccess() != null && !cat.getAccess().isRead() )
```
Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FormUtils.java]: Is it better with?

option.getAccess() == null || !option.getAccess().isRead() )

Comment (vietnguyen) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FormUtils.java]: This FormUtils.fromDataSet is also called by DataSetController, and currently it's not checking for user authorities and also organisationUnit access. 

So if those authorities = null means that it should not check and everything should be returned..

Comment (vietnguyen) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FormUtils.java]: @larshelge

Comment (larshelge) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FormUtils.java]: OK good

**Pull request merged**